### ISSUE TITLE
problem about the grad of dcn's weight

### ISSUE THREAD

**Issue by zyxunh** [chengdazhi/Deformable-Convolution-V2-PyTorch#3]
I test the grad of dcn by setting mask to 1 and offset to 0, so the result should be the same as nn.Conv2d. But I get a confusing result. The test code is:
    from modules.modulated_dcn import ModulatedDeformConv
    input = torch.ones([1,1,10,10],dtype=torch.float).cuda()
    offset= torch.zeros([1,9*2,10,10],dtype=torch.float).cuda()
    mask= torch.ones([1,9,10,10],dtype=torch.float).cuda()

    dcn=ModulatedDeformConv(in_channels=1, out_channels=1,
                 kernel_size=3, stride=1, padding=1,deformable_groups=1)
    dcn=dcn.cuda()
    dcn.weight.data[...]=1#
    out=dcn(input,offset,mask)
    loss=out.sum()
    loss.backward()
    print(dcn.weight.grad)
    cn=nn.Conv2d(1,1,
                kernel_size=3,
                stride=(1, 1),
                padding=(1, 1),
                bias=False)
    cn=cn.cuda()
    cn.weight.data[...]=1
    out=cn(input)
    loss=out.sum()
    loss.backward()
    print(cn.weight.grad)

and the outputs are:
tensor([[[[ 0.,  0.,  0.],
          [ 0.,  1.,  1.],
          [ 0.,  1.,  1.]]]], device='cuda:0')
tensor([[[[  81.,   90.,   81.],
          [  90.,  100.,   90.],
          [  81.,   90.,   81.]]]], device='cuda:0')

If there is error in my code?

**Comment by chengdazhi** [comment]
Hi, I can reproduce your results, however, if I call backward on out `out.backward(torch.ones_like(out))` instead of loss 'loss=out.sum();loss.backward()', I can get correct gradients. I will look further into this.
